### PR TITLE
Problem: COSS-formatted URLs were broken

### PR DESCRIPTION
**Pull request by yrashk**
Solution: Use a COSS plugin to generate all URLs

### CODE REVIEW

**Comment by yrashk** [comment]
For example, we only had /42 as available path, but not /spec:42 or /spec:42/C4

This plugin that I just developed, fixes the problem.

**Comment by yrashk** [comment]
Please hold on, some gitbook hosting-related issues

**Comment by yrashk** [comment]
The summary of the hosting-related issue:

When viewing RFCs through `gitbook serve` server, it will append necessary trailing slashes to URLs (like /spec:28**/**). However, on gitbook.com, that's not happening. Waiting for them to respond.

**Comment by yrashk** [comment]
Still haven't heard from GitBook guys. Maybe @SamyPesse can help figuring this issue out?

**Comment by SamyPesse** [comment]
@yrashk not sure I understand the problem, I'll search for your support ticket (sorry that I missed it).

Can you explain in details what is the expected result and the current result?

**Comment by yrashk** [comment]
@SamyPesse I can show the behaviour on another book built using the same approach.

http://rfc.unprotocols.org/spec:1/ works but http://rfc.unprotocols.org/spec:1 doesn't (404). However, locally, when using `gitbook serve`, /spec:1 works (albeit by redirecting to /spec:1/, which is not perfect, but acceptable). ZeroMQ RFC links are all over the place without the trailing slash so we can't just accept the current gitbook.com serving behaviour as it'll break things. Besides, in general, I'd think that allowing both /spec:1/ and /spec:1 would be best.

Does this help?

**Comment by somdoron** [comment]
@yrashk @SamyPesse any update on this?

**Comment by SamyPesse** [comment]
@yrashk @somdoron It should be good now.

File: book.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
     title: 'ZeroMQ RFCs',
-    gitbook: '3.0.0-pre.15'
+    gitbook: '3.0.0-pre.15',
```
Comment (SamyPesse) [inline book.js]: You should replace this line by `>=3.0.0`

Comment (yrashk) [inline book.js]: Is it guaranteed to pick up the latest beta both on gitbook.com and when using gitbook locally?

**Comment by yrashk** [comment]
It's good to merge now, it seems. It works for Unprotocols, see http://rfc.unprotocols.org/spec:1/C4 (it will append the trailing slash) or http://rfc.unprotocols.org/spec:1

We can address the gitbook version issue in a separate PR to get this through faster.

// cc @somdoron @hintjens

**Comment by yrashk** [comment]
@SamyPesse thank you so much for your help! ❤️

File: book.js (hunk 1 above)
Comment (SamyPesse) [inline book.js]: We release GitBook 3.0.0 as stable (not yet announced because we still want to investigate potential issues), so yes it'll work.

**Comment by SamyPesse** [comment]
You're welcome 🍻

**Comment by yrashk** [comment]
@SamyPesse this is not happening on rfc.unprotocols.org but happens for ZeroMQ, most every time I click on a link in the ToC, the chapter loads but I also get a "Error processing comments: error" message box. Is it something I could have done incorrectly?

**Comment by yrashk** [comment]
@somdoron Barring this  "Error processing comments: error" message problem, we are ready to flip the CNAME over to gitbook. This problem is not present on http://rfc.unprotocols.org. Not sure what's causing it yet.

**Comment by SamyPesse** [comment]
What is the url of the page with this error?

**Comment by yrashk** [comment]
@SamyPesse go to https://zeromq.gitbooks.io/rfc/content and click on 11/MTL chapter on the left.  Going to 11/MTL's chapter directly (via a direct link) does not reproduce the error.

**Comment by yrashk** [comment]
There's a chance that this is somehow related to being hosted on gitbooks subdomain with the /rfc/content root. After all, both http://rfc.unprotocols.org and http://rfc.eventsourcing.com work just fine.

**Comment by SamyPesse** [comment]
Yes, we are going to fix it in the next GitBook release, the issue with the comment plugin come from the fact that there is a redirection in the AJAX navigation.

It'll probably be fixed tomorrow.

**Comment by yrashk** [comment]
@SamyPesse If we publish it to rfc.zeromq.org CNAME, will the problem go away by any chance?

**Comment by SamyPesse** [comment]
yes, it'll probably go away, since the browser will resolve `rfc.zeromq.org/../gitbook/api/` as `rfc.zeromq.org/gitbook/api/` which will be correct.

**Comment by yrashk** [comment]
then I guess we're good to go with the CNAME change, both in gitbooks.com UI and DNS.

// cc @somdoron

**Comment by SamyPesse** [comment]
@yrashk We fixed the issue.

**Pull request merged**